### PR TITLE
Add contributing guide and issue/PR templates

### PR DESCRIPTION
**Pull request by sagikazarmark**
## ToDo

- [x] Update issue templates

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:6]:
```suggestion
The goal of this guide is to ensure making contributions goes smoothly. If you have trouble with this process, please help us to improve it by proposing changes.
```

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:18]:
```suggestion
If you think a feature is missing, please [make a suggestion](https://github.com/dexidp/helm-charts/issues/new?assignees=&labels=kind%2Fenhancement&template=feature_request.md) or, even better, implement it.
```

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:11]:
```suggestion
If you find a bug or a documentation issue, please [report it](https://github.com/dexidp/helm-charts/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.md) or, even better, fix it.
```

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:20]:
```suggestion
If you do report it, please describe your use case concretely and with as much details as possible.
```

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:31]: This is the first time `CI` is used, use the full form before abbreviating it.
```suggestion
and lets the continuous integration (CI) system, in this case [GitHub Actions](https://docs.github.com/en/actions), run automated checks to make sure the proposed change doesn't break anything.
```

Do we need to refer to GitHub actions here?

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:53]:
```suggestion
git clone git@github.com:YOUR-GITHUB-USERNAME/helm-charts.git
```

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:45]: Is there any specifics around how the clusters need to be set up? Or are the getting started instructions for both of these options sufficient to get set up for this process?

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:145]: Do we need to mention potential future commit message usage? As a future change to the release process would require an edit here, I'd drop it for now and add the info when the feature is implemented.

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [inline .github/CONTRIBUTING.md:243]: My opinion: I'd drop the "Eventually your changes will be merged." line -- I appreciate that you're trying to encourage the audience, but it may not actually be merged in the end.

```suggestion
Congratulations on the contribution and thanks! 🎉👍
```

**Review by nate-double-u — COMMENTED** [review]
@sagikazarmark I've done a copy pass on the text, and have left some suggestions in line. I haven't yet done a line by line test of the Contribution process. I'd especially like to see a double check of the code/terminal snippets explaining how to set things up and test before this PR gets approved (I'll see if I can run through it this week).

**Comment by sagikazarmark** [comment]
> I'll see if I can run through it this week

That would be great. I actually tested the commands while writing the guide, but haven't gone through the whole process.

Thanks for the review!

File: .github/CONTRIBUTING.md
Comment (sagikazarmark) [reply .github/CONTRIBUTING.md:45]: > Is there any specifics around how the clusters need to be set up?

Not really. It might be worth mentioning that the chart either needs an ingress controller to expose the service publicly, or it can also be tested using `kubectl port-forward`.

I'm not sure how much detail we should be going into.

In order to test the chart the contributor first needs to understand how it works, including the potential requirement for an ingress controller.

File: .github/CONTRIBUTING.md
Comment (sagikazarmark) [reply .github/CONTRIBUTING.md:145]: I don't mind dropping it. The reason I mentioned future use cases is because it's hard to reason about using conventional commits if we don't actually make use of it.

File: .github/CONTRIBUTING.md
Comment (sagikazarmark) [reply .github/CONTRIBUTING.md:243]: Yeah, I wasn't sure about that part. :)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (sagikazarmark) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: I think this is a copy-paste from the Dex repo. Signing a DCO means signing off all commits in the PR. Not sure how we can improve it.

**Review by nabokihms — DISMISSED** [review]
Looks good. I did not manage to find anything odd.

File: .github/CONTRIBUTING.md
Comment (sagikazarmark) [reply .github/CONTRIBUTING.md:209]: It's already checked. :)

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (sagikazarmark) [reply .github/ISSUE_TEMPLATE/bug_report.md:1]: See todos

**Comment by justaugustus** [comment]
Can review in the next few days!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (nate-double-u) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: I wouldn't let it be a blocker then :)

File: .github/CONTRIBUTING.md
Comment (nate-double-u) [reply .github/CONTRIBUTING.md:145]: That's enough of an argument for me if you'd like to keep it in :)

**Review by justaugustus — DISMISSED** [review]
@sagikazarmark -- I may have some microscopic nits, but going to defer them and let's iterate. Really great!

Let's fix the DCO failure in https://github.com/dexidp/helm-charts/pull/42/commits/2c04f90f23baddd522b9df2853dc8d87c0925e60 and merge.

**Comment by sagikazarmark** [comment]
Thanks for the review everyone. I slapped in some issue templates as well. Feel free to comment here or open an issue if you see something odd.

**Pull request merged**
